site/content/docs/Encryption/_index.md

```diff
@@ -0,0 +1,104 @@
+---
+title: "Encryption"
+linkTitle: "Encryption"
+weight: 4
+---
+
+Kopia uses a standard envelope encryption technique to de-couple the repository passphrase from the keys used for encrypting and authenticating the contents of the repository.
+
+Each repository has a format blob containing configuration parameters for the repository. The format blob itself is serialized and persistent as JSON, such as the one below.
+
+not a good random initial value
+
+```json
+{
+  "tool": "https://github.com/kopia/kopia",
+  "buildVersion": "v0.3.0",
+  "buildInfo": "unknown",
+  "uniqueID": "bm90IGEgZ29vZCByYW5kb20gaW5pdGlhbCB2YWx1ZQ==",
+  "keyAlgo": "scrypt-65536-8-1",
+  "version": "1",
+  "encryption": "AES256_GCM",
+  "encryptedBlockFormat": "ZSxqt/gXwVPS6pNvFZOOHlCKr18TmziuUnN8nnuvwQ/+mjbcvEHUfKS11RJl/sWrIOyiYqpSwAZt
+BzOxQAUWQt7vHc2ZT4Y75ODrPHhzd0CcBlqHa3HSx8pxIXqpgMy5K3xDvIkBN1qdb/cTyU5s9lZ2
+J+rBh2CGQ3phIlKFCCuS3lgB+rnYRrExGg5rm4BUmZZWHfBPQ7QiOJNg86PK+n///dejsAA/+FBj
+qIODf7Gr3ppaGZeAHYJuz0BkRSPC4XIAgFj1yPo="
+}
+```
+
+The corresponding Go struct is the following:
+
+```go
+type formatBlob struct {
+	Tool         string `json:"tool"`
+	BuildVersion string `json:"buildVersion"`
+	BuildInfo    string `json:"buildInfo"`
+
+	UniqueID               []byte `json:"uniqueID"`
+	KeyDerivationAlgorithm string `json:"keyAlgo"`
+
+	Version              string                  `json:"version"`
+	EncryptionAlgorithm  string                  `json:"encryption"`
+	EncryptedFormatBytes []byte                  `json:"encryptedBlockFormat,omitempty"`
+	UnencryptedFormat    *repositoryObjectFormat `json:"blockFormat,omitempty"`
+}
+```
+
+* `Version` identifies the version of the format blob.
+* The `tool` and `buildInfo` fields are informational
+* `buildVersion` is the version of Kopia, which is also indirectly used as the repository format version.
+* `UniqueID` is a randomly generated identifier for the repository. This is also used as the input for various encryption operations.
+* `keyAlgo` identifies the password-based key derivation function (PBKDF). Only _scrypt_ with the currently recommended cost parameters (N=65536, r=8, p=1) is supported at the moment. The main purpose of this field is to be able to change and extend the key derivation algorithm in the future. For example, by increasing the cost parameters for _scrypt_ or using a different algorithm altogether.
+* `encryption` identifies the encryption algorithm that was used to encrypt the encryptedBlockFormat field.
+* `encryptedBlockFormat` is a ciphertext containing among others, the encryption secrets and parameters used for encrypting the repository content. Below is additional information about its plaintext content and how it is encrypted.
```

Comment:
might be worth mentioning that `blockFormat` can alternatively be specified

Reply:
Makes sense. Any suggestions for linking, maybe to command line options?

Reply:
OK, I guess I understand your comment better now, the unencrypted contents can be specified / inlined in the format blob. Got it.